keel-ec2-api/src/main/kotlin/com/netflix/spinnaker/keel/api/ec2/ApplicationLoadBalancerSpec.kt

```diff
@@ -39,10 +36,18 @@ data class ApplicationLoadBalancerSpec(
   data class Listener(
     val port: Int,
     val protocol: String,
-    val certificateArn: String?,
+    val certificate: String? = null,
     val rules: Set<Rule> = emptySet(),
     val defaultActions: Set<Action> = emptySet()
-  )
+  ) {
+    init {
+      if (protocol == "HTTPS") {
+        requireNotNull(certificate) {
+          "HTTPS listeners must specify a certificate"
+        }
+      }
+    }
+  }
```

Comment:
Change to the listener spec 1) field is the certificate name rather than ARN, 2) certificate is required if using HTTPS